src/backend/marsha/core/tests/test_models_video.py

```diff
@@ -43,6 +52,20 @@ def test_models_video_fields_lti_id_unique(self):
         video.delete(force_policy=SOFT_DELETE_CASCADE)
         VideoFactory(lti_id=video.lti_id, playlist=video.playlist)
 
+    def test_models_video_live_type_check_constraint(self):
+        """Videos live_type and live_state fields must be set together."""
+        with self.assertRaises(IntegrityError):
+            with transaction.atomic():
+                VideoFactory(
+                    live_state=random.choice([choice[0] for choice in LIVE_CHOICES])
+                )
+
+        with self.assertRaises(IntegrityError):
+            with transaction.atomic():
+                VideoFactory(
+                    live_type=random.choice([choice[0] for choice in LIVE_TYPE_CHOICES])
+                )
+
```

Comment:
You don't need the atomic wrapper but can do 2 tests:
- a live state should not be set without a live type
- a live type should not be set without a live state

Reply:
Done